x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/index.ts

```diff
@@ -492,6 +492,39 @@ export class KnowledgeBaseService {
     };
   };
 
+  getInstructions = async (): Promise<UserInstruction[]> => {
+    try {
+      const response = await this.dependencies.esClient.search<KnowledgeBaseEntry>({
+        index: this.dependencies.resources.aliases.kb,
+        query: {
+          bool: {
+            must: [
+              {
+                term: {
+                  'labels.category.keyword': {
+                    value: 'instruction',
+                  },
```

Comment:
I'm surprised we need `labels.category.keyword` here? is it not indexed as a keyword by default?

Reply:
When I checked my mappings looked like:
`labels > category`
`labels > category > keyword`

The mappings set it up as:
` labels: dynamic`

So I'm guessing ES guesses it's type `text` since it's a string but then also add a multifield with type `keyword`.

Reply:
that makes sense, we do have a terms query on `labels.category` elsewhere though (instead of `labels.category.keyword`, so I think we should be consistent here.